### PR TITLE
Convert protelis-lang from rep/nbr to share

### PR DESCRIPTION
**Pull request by jakebeal**
I have performed an initial conversion of all of protelis-lang from rep/nbr to the new share command.
All tests are passing except for three that are testing `circularDependency` from the `graph.pt` library.

Can you please assess whether this indicates that:
1. This indicates an issue in the implementation of `share`, or
2. This is an issue with the tests themselves or, 
3. There is an issue in the `graph.pt` update that should be rolled back or fixed

### CODE REVIEW

**Comment by DanySK** [comment]
I never touched graph.pt myselft, I wonder if @w4bo made it, and if he has any suggestion

**Comment by jakebeal** [comment]
Looks like svn:blame attributes all of it to @w4bo .

File: protelis/protelis-lang/src/main/protelis/protelis/coord/accumulation.pt [hunk 1]
```diff
@@ -59,7 +59,7 @@ public def boundAverage(sink, local, range, null) {
  * @return          T, aggregated value
  */
 public def C(potential, reduce, local, null) {
-    rep (v <- local) {
+    share (v,nbr_v <- local) {
```
Comment (DanySK) [inline protelis/protelis-lang/src/main/protelis/protelis/coord/accumulation.pt]: Do we need  `v`? Can we drop it and only use `nbr_v`? (even better, remove `nbr_v` and rename its usages to `v`)

**Comment by jakebeal** [comment]
I found the issue in `graph.pt`: looks like there was an exchanged argument in the original, and the bug wasn't tickled until I made the share conversion

File: protelis/protelis-lang/src/main/protelis/protelis/coord/accumulation.pt (hunk 1 above)
Comment (jakebeal) [reply protelis/protelis-lang/src/main/protelis/protelis/coord/accumulation.pt]: Done.  That was a left-over artifact from when I was figuring out the that the subfunction `v` wasn't the same `v`

**Comment by jakebeal** [comment]
Found a bug in `isLeaf`: I don't know how it ever passed tests in the past.

**Comment by jakebeal** [comment]
@DanySK Is there a reason this is not being able to be integrated, or can we merge?

**Comment by DanySK** [comment]
I made a number of style changes, if I have not broken anything in the process, for me it's good to go!

**Pull request merged**